resetcloser.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"io"
+)
+
+type ResetCloser struct {
+	c io.Closer
```

Comment:
That clearly doesn't work (non-pointer receiver) because you need to modify the contents of the struct. I've updated my review comment code to reflect this.